python/TestHarness/testers/SchemaDiff.py

```diff
@@ -0,0 +1,165 @@
+            return output
+
+        # Don't run on scaled tests
+        if options.scaling and specs['scale_refine']:
+            return output
+
+        # Loop over every file
+        for file in specs['schemadiff']:
+            gold_file = specs['gold_file'] if specs['gold_file'] else file
+            if not os.path.exists(os.path.join(self.getTestDir(), specs['gold_dir'], gold_file)):
+                output += "File Not Found: " + os.path.join(self.getTestDir(), specs['gold_dir'], file)
+                self.setStatus(self.fail, 'MISSING GOLD FILE')
+                break
+
+            # Perform diff
+            else:
+                gold = os.path.join(self.getTestDir(), specs['gold_dir'], gold_file)
+                test = os.path.join(self.getTestDir(), file)
+                # We always ignore the header_type attribute, since it was
+                # introduced in VTK 7 and doesn't seem to be important as
+                # far as Paraview is concerned.
+                specs['ignored_items'].append('header_type')
+
+                gold_dict = self.try_load(gold)
+                test_dict = self.try_load(test)
+
+                if isinstance(gold_dict, Exception):
+                    output += "Gold Schema File Failed To Load: "+gold+"\n"
+                    output += "Gold Schema Exception: " + str(gold_dict) + "\n"
+                    self.setStatus(self.fail, 'LOAD FAILED')
+
+                if isinstance(test_dict, Exception):
+                    output += "Test Schema File Failed To Load: "+test+"\n"
+                    output += "Test Schema Exception: " + str(test_dict) + "\n"
+                    self.setStatus(self.fail, 'LOAD FAILED')
+
+                #Break after testing both to provide both errors in the log if both are applicable
+                if isinstance(test_dict, Exception) or isinstance(gold_dict, Exception):
+                    break
+
+                # Perform the diff.
+                diff = self.do_deepdiff(gold_dict, test_dict, specs['rel_err'], specs['abs_zero'], specs['ignored_items'])
+                if diff:
+                    output += "Schema difference detected.\nFile 1: " + gold + "\nFile 2: " + test + "\nErrors:\n"
+                    output += diff
+                    self.setStatus(self.diff, 'SCHEMADIFF')
+                    break
+        return output
+
+    def do_deepdiff(self,orig, comp, rel_err, abs_zero, exclude_values:list=None):
+        import deepdiff
+        from deepdiff.operator import BaseOperator
+        class testcompare(BaseOperator):
+            def __init__(self, rel_err,abs_zero,types,regex_paths=None):
+                self.rel_err = rel_err
+                self.abs_zero = abs_zero
+                #next two members are necessary for deepdiff constructor to work
+                self.regex_paths = regex_paths
+                self.types = types
+
+            def give_up_diffing(self,level, diff_instance):
```

Comment:
Where is this actually called?

Reply:
It's an overriden internal method of DeepDiff, its ran on every diff, its used to check if it should attempt a diff or not